Competitive_Programming/C++/Searching & Sorting/Binary Search/binary_search_documentation.md

```diff
@@ -0,0 +1,241 @@
+
+# BINARY SEARCH 
+
+Binary search is also known as half-interval search. It is a searching technique in a sorted array by repeatedly dividing search intervals in half. It is an efficient algorithm to find the element efficiently. Repeatedly check until the value is found or the interval is empty.
+
+
+
+
+## Steps To Implement Binary Search
+Let us assume there is an array A and an element x.
+We have to search throughout the array and we need to find element x inside the array.
+If not present print -1 or print the position of the array.
+
+
+---
+<p align="center">
+  <img width="460" height="300" src="https://miro.medium.com/max/247/1*fMQ2cLdIEr3DoHitYyiSsg.png">
```

Comment:
Can you please use a clearer image? And please give image credit if you are using an image from a different source.